qubes/storage/__init__.py

```diff
@@ -607,15 +697,33 @@ async def remove(self):
         except (IOError, OSError) as e:
             self.vm.log.exception("Failed to remove some volume", e)
 
+    def block_devices(self):
+        """ Return all :py:class:`qubes.storage.BlockDevice` for current domain
+        for serialization in the libvirt XML template as <disk>.
+        """
+        for v in self.vm.volumes.values():
+            block_dev = v.block_device()
+            if block_dev is not None:
+                if v.ephemeral:
+                    yield v.make_encrypted_device(block_dev, self.vm.name)
+                else:
+                    yield block_dev
+
     async def start(self):
         ''' Execute the start method on each volume '''
         await qubes.utils.void_coros_maybe(
-            vol.start() for vol in self.vm.volumes.values())
+            # pylint: disable=line-too-long
+            (vol.start_encrypted(vol.encrypted_volume_path(self.vm.name, name))
+             if vol.ephemeral else vol.start())
+            for name, vol in self.vm.volumes)
```

Comment:
You need this:
```suggestion
            for name, vol in self.vm.volumes.items())
```
And same in `stop()`.

Reply:
:woman_facepalming: Oops!